README.md

```diff
@@ -0,0 +1,60 @@
+# tool-upgrade-go
+
+A library for golang that helps upgrade tools to newer versions with ease.
+
+## Usage
+
+### Repository structure
+
+```
+# over-simplified example of directory and file structure required
+# mycompany.com web server directory or artifact repository
+
+# binaries must be suffixed with _${GOOS}_${GOARCH}
+% find path/to/dist
+path/to/dist/coolTool_darwin_amd64
+path/to/dist/coolTool_linux_amd64
+path/to/dist/coolTool_windows_amd64
+
+% cat path/to/coolTool/.version-latest
+1.2.2
+
+# copy new binaries into place
+% mkdir -p path/to/coolTool/1.2.3
+% cp path/to/dist/coolTool_* path/to/coolTool/1.2.3
+
+# update version channel
+% echo "1.2.3" > path/to/coolTool/.version-latest
+```
+
+### Check for updates and upgrade if needed
+
+`tool-upgrade-go` will upgrade to the version available in the version channel.
+It leverages `GOOS` and `GOARCH` of the current runtime to determine which
+platform architecture to download for the upgrade. In practice, this is often
+done using a CLI subcommand.
+
+```
+import "github.com/VeritasOS/tool-upgrade-go"
```

Comment:
Does the following example mean an automatic upgrade?  Would it be worthwhile to give an example where an upgrade command is passed?

Reply:
Yes, the example would be an automatic upgrade. I think showing use of the library's API is enough, at least initially. I think there are some improvements to be made to the interface, so I hesitate to build out more examples before we see those changes shake out.